furusiya/model/components/ai/monster.py

```diff
@@ -20,17 +22,26 @@ def take_turn(self):
                 monster.move_towards(Game.player.x, Game.player.y)
 
             # close enough, attack! (if the player is still alive.)
-            elif Game.player.fighter.hp > 0:
-                monster.fighter.attack(Game.player)
+            elif Game.player.get_component(Fighter).hp > 0:
+                monster.get_component(Fighter).attack(Game.player)
 
 
-class StunnedMonster:
+class TemporaryAI(AI):
```

Comment:
I don't like this. The reason I decided not to do this is the following workflow:

- Create a monster with a regular AI
- Stun it
- Confuse it before it gets un-stunned

In this case, you get a permastunned AI. Not at all expected or intended.

We should just attach the original/master/main AI to the game object instance, and refer back to that when effects wear off.  I am also not opposed to (but will not currently) implement a decorator/filter design pattern where status effects stack and interact with each other :D